heare/developer/agent.py

```diff
@@ -509,9 +509,16 @@ def run(
                             )
             elif final_message.stop_reason == "max_tokens":
                 user_interface.handle_assistant_message(
-                    "[bold red]Hit max tokens.[/bold red]"
+                    "[bold yellow]Hit max tokens. I'll continue from where I left off...[/bold yellow]"
                 )
 
+                # Add a continuation prompt to the tool result buffer
+                continuation_prompt = {
```

Comment:
I think this will break if in the middle of a tool_use block -- we want to continue the assistant turn -- can we just put the final message back into the tool_result_buffer instead of appending it to conversation history?